src/workers/newView.ts

```diff
@@ -19,60 +22,24 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
     !existing ||
     entity.timestamp.getTime() - existing.timestamp.getTime() > ONE_WEEK
   ) {
-    await repo.save(entity);
-    return true;
+    const viewRecord = await repo.save(entity);
+    return { view: viewRecord, isNewView: true };
   }
-  return false;
+  return { view: existing, isNewView: false };
 };
 
-const DEFAULT_TIMEZONE = 'UTC'; // in case user doesn't have a timezone set
-const INC_STREAK_QUERY = `
-WITH u AS (
-  SELECT "id",
-      /* Increment the current streak if
-        a) lastViewAt is NULL - this is a new user, we should start a new streak
-        b) the currentStreak is 0 - we should start a new streak
-        c) we didn't do it today already
-      */
-      ("lastViewAt" is NULL OR
-       "currentStreak" = 0 OR
-       DATE("lastViewAt" AT TIME ZONE COALESCE("timezone", $3)) <> DATE($2 AT TIME ZONE COALESCE("timezone", $3))
-      ) AS shouldIncrementStreak
-  FROM public.user AS users
-  INNER JOIN user_streak ON user_streak."userId" = users.id
-  WHERE users.id = $1
-),
-updated AS (
-  UPDATE user_streak AS us SET
-    "lastViewAt" = $2,
-    "updatedAt" = now(),
-    "currentStreak" = CASE WHEN shouldIncrementStreak THEN us."currentStreak" + 1 ELSE us."currentStreak" END,
-    "totalStreak" = CASE WHEN shouldIncrementStreak THEN us."totalStreak" + 1 ELSE us."totalStreak" END,
-    "maxStreak" = CASE WHEN shouldIncrementStreak THEN GREATEST(us."maxStreak", us."currentStreak" + 1) ELSE us."maxStreak" END
-  FROM u
-  WHERE us."userId" = u.id
-  RETURNING us."userId", "currentStreak"
-)
-/* Return the updated streak data */
-SELECT updated."currentStreak", u.shouldIncrementStreak AS "didIncrementStreak"
-FROM updated
-JOIN u ON TRUE;
-`;
-
 const incrementReadingStreak = async (
   con: DataSource,
-  data: DeepPartial<View>,
+  latestView: View,
 ): Promise<boolean> => {
   const users = con.getRepository(User);
   const repo = con.getRepository(UserStreak);
-  const { userId, timestamp } = data;
+  const { userId, timestamp } = latestView;
 
   if (!userId) {
     return false;
   }
 
-  const viewTime = timestamp ? new Date(timestamp as string) : new Date();
-
   // TODO: This code is temporary here for now because we didn't run the migration yet,
```

Comment:
Since we are touching the file, we can basically get rid of this logic now. The migration has been run, so all users will have a record in the `user_streak` table AFAIK.

Reply:
yep, already mentioned will be cleaning this up in a separate PR 🙏